### DISCUSSION TITLE
Vercel type Error for Api response

### DISCUSSION THREAD

**Discussion by MickyRajkumar** [vercel/next.js#55728] [Help]
### Summary

In my nextjs 13 api route.ts 

`export async function POST(req: NextRequest) {
const body = await req.json();
....
...
return new Response('Ok', { status: 200 });
}`
and It working fine locally there is no error even when build. But When I deployed on vercel I throw an error when build 
`Type error: Route "src/app/api/register-product/route.ts" has an invalid export:
 "Promise<Response | undefined>" is not a valid POST return type: 
Expected "Response | Promise<Response>", got "Promise<Response | undefined>". 
Expected "Promise<Response>", got "Promise<Response | undefined>". 
Expected "Response", got "Response | undefined". 
Expected "Response", got "undefined".`

and by default response type is Response: new (body?: BodyInit | null | undefined, init?: ResponseInit | undefined) => Response

### Example
![Screenshot 2023-09-21 165501](https://github.com/vercel/next.js/assets/74924600/d1760742-87b7-4cd0-af42-d4bc7822daca)

![Screenshot 2023-09-21 165212](https://github.com/vercel/next.js/assets/74924600/77f67100-8001-4177-adf3-16d5b0692c44)

**Comment by icyJoseph** [comment]
Have you tried with next@13.5.2? Or rather, what version are you using?

**Reply by MickyRajkumar** [reply]
I am using "next": "^13.4.12",

**Reply by icyJoseph** [reply]
Did you manage to fix it?
